services/filesstore/s3store.go

```diff
@@ -238,10 +238,15 @@ func (b *S3FileBackend) ListDirectory(path string) (*[]string, *model.AppError)
 	}
 
 	doneCh := make(chan struct{})
-
 	defer close(doneCh)
 
-	for object := range s3Clnt.ListObjects(b.bucket, path, false, doneCh) {
+	updatedPath := path
+	if !strings.HasSuffix(path, "/") && len(path) > 0 {
+		// s3Clnt returns only the path itself when "/" is not present
+		// appending "/" to make it consistent across all filesstores
+		updatedPath = path + "/"
```

Comment:
Any reason not to just mutate `path` directly?

Reply:
Habit developed from JS, usually refrain from updating passed args. Updated.